napari/components/viewer_model.py

```diff
@@ -352,20 +352,20 @@ def _sliced_extent_world(self) -> np.ndarray:
         -------
         sliced_extent_world : array, shape (2, D)
         """
-        if len(self.layers) == 0 and self.dims.ndim != 2:
```

Comment:
I don't understand why the second part of the condition is related to this PR. If you make this change, does the explanatory/implementation comment below need to change?

Reply:
I think this slipped through, but now that I'm reading it I don't understand this comment. do you? To me it looks like it's going to fail if `ndim = 2` and there are no layers.

Reply:
I think I understand the intent of the comment (about `dims.axis_labels`), but I'm not sure if that's actually the behavior right now - hopefully there's at least one test that covers that behavior if it's expected.

After a quick look, it looks like `LayerList._extent_world_augmented` handles the case when there are no layers such that this won't won't fail when `ndim == 2`.

Can take a deeper look if you don't have time.

Reply:
Turns out we actually need to do this check in `ViewerModel`, because we need access to the `Dims` in order to create extents with the correct dimensionality (this is only relevant in the case where `viewer.layers.ndim == 2 and viewer.dims.ndisplay == 3` where slicing would break).